packages/hash/frontend/src/components/hooks/blockProtocolFunctions/ontology/ontology-types-shim.ts

```diff
@@ -39,7 +40,7 @@ export type OntologyCallbacks = {
 
 /* Data type CRU */
 export type AggregateDataTypesMessageCallback = MessageCallback<
-  {},
+  EmptyObject,
```

Comment:
Shouldn't we allow specifying `constrainsValuesOn` here for non-primitive data types?

Reply:
They don't exist at the moment, so, no 😁 , there are tracked todo's elsewhere in the code to address this :)